### PR TITLE
Add date type support to dynamic control

### PR DESCRIPTION
**Pull request by lapostoj**
Trying to add support for a date type in the dynamic control form.

Also reorganizing a bit the example page menu to make it easier to find the components.

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -41,26 +41,32 @@ <h3 class="panel-title">Form Styling &amp; Validation</h3>
...
 						<li><a href="#radio">Radio</a></li>
-						<li><a href="#requirements-form">Requirements form</a></li>
 						<li><a href="#select">Select</a></li>
-						<li><a href="#tool-tip">Tool tip</a></li>
 						<li><a href="#upload">Upload</a></li>
-						<li><a href="#validation">Validation</a></li>
+						<li><a href="#currency-input">Currency input</a></li>
+						<li><a href="#amount-currency-select">Amount currency select</a></li>
+						<li><a href="#dynamic">Dynamic control</a></li>
+						<li><a href="#requirements-form">Requirements form</a></li>
```
Comment (StevePole) [inline index.html]: I'm ok with separating, but they were also in alphabetical order before!

File: partials/tw-dynamic-control.html [hunk 2]
```diff
@@ -178,14 +182,54 @@ <h5 class="page-header">
...
+			<div class="form-group"
+				ng-if="vm.dynamic.type !== 'radio' && vm.dynamic.type !== 'date'">
 				<label class="control-label">
 					Placeholder
 				</label>
 				<input type="text"
 					class="form-control"
 					ng-model="vm.dynamic.placeholder"
-					ng-init="vm.dynamic.placeholder = 'Example'" />
+					ng-init="vm.dynamic.placeholder='Example'" />
```
Comment (StevePole) [inline partials/tw-dynamic-control.html]: Why?

File: angular/components/tw-dynamic-form-control/tw-dynamic-form-control.directive.js [hunk 3]
```diff
@@ -20,6 +20,7 @@
 				id: "@",
 				placeholder: "@",
 				step: "@",
+				locale: "@",
```
Comment (StevePole) [inline angular/components/tw-dynamic-form-control/tw-dynamic-form-control.directive.js]: Are you going to add this to the requirements form separately?  I suspect you also need to add it there.

File: partials/tw-dynamic-control.html (hunk 2 above)
Comment (lapostoj) [inline partials/tw-dynamic-control.html]: Thought I saw others that way, maybe not...

File: index.html (hunk 1 above)
Comment (lapostoj) [inline index.html]: Oh didn't notice! Yup can go back to that!

File: angular/components/tw-dynamic-form-control/tw-dynamic-form-control.directive.js (hunk 3 above)
Comment (lapostoj) [inline angular/components/tw-dynamic-form-control/tw-dynamic-form-control.directive.js]: Good point, adding it

**Pull request merged**